pkg/cloudprovider/registry.go

```diff
@@ -0,0 +1,71 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloudprovider
+
+import (
+	"sync"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+type ResourceFlags byte
+
+const (
+	// ResourceFlagNone is the default settings for resources in which instances prefer to scale up vs out and the resource
+	// type is not considered exotic
+	ResourceFlagNone ResourceFlags = iota
+
+	// ResourceFlagScaleOut indicates that if this resource is exhausted, Karpenter will launch a new instance
+	// instead of choosing a larger capacity instance size. This is used for resources in which its cheaper or more
+	// effective to scale horizontally (e.g. AWS ENIs) than scaling up.
+	// ResourceFlagScaleOut
+	// TODO(todd): implement once bin-packing and scheduling are combined
+
+	// ResourceFlagExotic indicates that instances of this resource should not be considered unless a pod specifically
+	// requests this resource type.  This is useful for preventing non-GPU workloads from possibly scheduling to more
+	// expensive GPU instance type, or from causing a GPU instance type to scale up to the next larger type.
+	ResourceFlagExotic
+)
+
+var mu sync.RWMutex
+var resourceInfo = map[v1.ResourceName]ResourceFlags{}
```

Comment:
Thoughts on making this public?

Reply:
There may be a reason to change it some day? I don't think making it public provides much benefit, but it's a toss up either way.

Reply:
It enables us to remove the setters/getters in favor of direct access. In my understanding, go style typically leans towards direct access rather than setters/getters unless they're doing something specific. In this case, you were wrapping with a mutex, but if we remove the thread safety, I don't see the value anymore.

Reply:
sgtm